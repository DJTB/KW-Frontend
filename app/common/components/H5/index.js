import styled from 'styled-components';
import { epsilon, headingRhythm } from 'common/styles/typography';

const H5 = styled.h5`
  ${epsilon}
  ${headingRhythm}
`;

export default H5;
