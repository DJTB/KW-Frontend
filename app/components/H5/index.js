import styled from 'styled-components';
import { fluidType } from 'shared/styles/utils';
import { minFontSize, mobileMod, desktopMod } from 'shared/styles/sizing';

// Modular scale
const mobileSize = +((mobileMod ** 1.5) * minFontSize).toPrecision(3);
const desktopSize = +((desktopMod ** 1.5) * minFontSize).toPrecision(3);

const H5 = styled.h5`
  ${fluidType(mobileSize, desktopSize)}
`;

export default H5;
