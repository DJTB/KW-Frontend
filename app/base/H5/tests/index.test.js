import 'jest-styled-components';
import { shallow } from 'enzyme';
import React from 'react';

import H5 from '../index';

describe('<H5 />', () => {
  it('should render a prop', () => {
    const id = 'testId';
    const renderedComponent = shallow(<H5 id={id} />);
    expect(renderedComponent.prop('id')).toEqual(id);
  });

  it('should match styling snapshot', () => {
    expect(shallow(<H5>text</H5>)).toMatchStyledComponentsSnapshot();
  });
});
