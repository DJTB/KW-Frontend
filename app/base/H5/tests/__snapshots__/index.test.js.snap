// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H5 /> should match styling snapshot 1`] = `
.eSbGZf {
  font-size: 15.3px;
  padding: 0.28125rem;
  margin-top: 0;
  margin-bottom: 0;
  line-height: 1.2;
  -webkit-letter-spacing: -1px;
  -moz-letter-spacing: -1px;
  -ms-letter-spacing: -1px;
  letter-spacing: -1px;
}

.eSbGZf:not(:first-of-type) {
  margin-top: 0.3999em;
  margin-bottom: 0;
}

@media (min-width: 600px) and (max-width: 1800px) {
  .eSbGZf {
    font-size: calc(15.3px + (17.4 - 15.3) * ( (100vw - 600px) / ( 1800 - 600) ));
  }
}

@media (min-width: 1800px) {
  .eSbGZf {
    font-size: 17.4px;
  }
}

@media (min-width: 600px) {
  .eSbGZf {
    padding: 0.33325rem;
  }
}

<h5
  className="sc-bdVaJa eSbGZf"
>
  text
</h5>
`;
